include/openPMD/IO/ADIOS/ADIOS2IOHandler.hpp

```diff
@@ -48,6 +38,17 @@
 #   include <mpi.h>
 #endif
 
+#include <array>
+#include <future>
+#include <memory> // shared_ptr
+#include <string>
+#include <utility> // pair
+#include <vector>
+
+#include "openPMD/auxiliary/JSON.hpp"
+#include <nlohmann/json.hpp>
```

Comment:
this should be before the std lib includes

Reply:
Is it [better now](https://github.com/openPMD/openPMD-api/blob/61be5f724229a056da50c1959519016eb3f7fbc2/include/openPMD/IO/ADIOS/ADIOS2IOHandler.hpp#L23)?

Reply:
Not 100% yet. I would
- prefix `ADIOS2FilePosition.hpp` with the usual full path
- include `openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp` before third party libs

```C++
#include "openPMD/config.hpp"
#include "openPMD/IO/AbstractIOHandler.hpp"
#include "openPMD/IO/AbstractIOHandlerImpl.hpp"
#include "openPMD/IO/AbstractIOHandlerImplCommon.hpp"
#include "openPMD/IO/IOTask.hpp"
#include "openPMD/IO/InvalidatableFile.hpp"
#include "openPMD/IO/ADIOS/ADIOS2FilePosition.hpp"
#include "openPMD/auxiliary/JSON.hpp"
#include "openPMD/backend/Writable.hpp"

#if openPMD_HAVE_ADIOS2
#   include "openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp"
#   include <adios2.h>
#endif

#include <nlohmann/json.hpp>

#if openPMD_HAVE_MPI
#   include <mpi.h>
#endif

#include <array>
// ...
```